js/source/tile.js

```diff
@@ -129,11 +129,11 @@ class Tile {
     }
 
     redoPlacement(source) {
-        if (source.type !== 'vector' && source.type !== 'geojson') {
+        if ((source.type !== 'vector' && source.type !== 'geojson') ||
+            !this.hasData() || !this.collisionTile) {
```

Comment:
Can you explain how this avoids the issue? It looks like checking `!this.collisionTile` will still prevent reaching the `this.redoWhenDone = true;` case when `redoPlacement` is called while the tile is initially loading.